exchange/exchange.go

```diff
@@ -1300,7 +1314,25 @@ func buildStoredAuctionResponse(storedAuctionResponses map[string]json.RawMessag
 			//set imp id from request
 			for i := range seat.Bid {
 				seat.Bid[i].ImpID = impId
-				bidsToAdd = append(bidsToAdd, &entities.PbsOrtbBid{Bid: &seat.Bid[i]})
+				mType := seat.Bid[i].MType
+				var bidType openrtb_ext.BidType
+				if mType > 0 {
+					if mType <= 4 {
+						//1 = Banner, 2 = Video,  3 = Audio, 4 = Native
+						bidType = openrtb_ext.BidTypes()[mType-1]
+					} else {
+						return nil, nil, nil, &errortypes.BadServerResponse{
+							Message: fmt.Sprintf("Failed to parse bid mType for impression \"%s\"", seat.Bid[i].ImpID),
+						}
+					}
+				} else {
+					var err error
+					bidType, err = getPrebidMediaTypeForBid(seat.Bid[i])
+					if err != nil {
+						return nil, nil, nil, err
```

Comment:
This error case has no test coverage.

Reply:
Added. 
Are we ok to return error as is from `ParseBidType`? Or is it better convert it to `BadServerResponse` for consistency?

Reply:
I don't have a strong opinion on this. I do like consistency and support you on that.

Reply:
Discussed offline. No changes needed.